README.md

```diff
@@ -17,76 +16,119 @@ _A table of contents to help people find things_
 - [Licence](#licence)
 
 ## Usage
-_Whatever usage instructions your component has. We've broken this down by Markup, JavaScript and Sass, but it depends how complex your component is._
+`o-message` uses Sass and Javascript to show and hide a message component.  
+It can be initialised declaratively if markup is provided on the page, or it can be initialised imperatively when using the [manual build process](http://origami.ft.com/docs/developer-guide/modules/building-modules/).
+
+By default, `o-message` initialises an alert message, which provides information in response to a user action. It currently has three states, 'success', 'neutral' and 'error', and relies on the markup (or configuration) to determine  other variants of behaviour and style.
+
 
 ### Markup
 
-_Common templating can go here, especially if there is only one template, but people can always check the demos for more._
+This is an example of the declarative way of instantiating an error message that spans **across a viewport**.
+
+```html
+<div class="o-message o-message--alert o-message--error" data-o-component="o-message">
+	<div class="o-message__container">
+		<div class="o-message__content">
+			<p class="o-message__highlight">Something went wrong!
+				<span class="o-message__detail">The quick brown fox did not jump over the lazy dogs.</span>
+			</p>
+			<div class="o-message__actions">
+				<a href="#" class="o-message__action--primary">Button</a>
+				<a href="#" class="o-message__action--secondary">Text link</a>
+			</div>
+		</div>
+	</div>
+</div>
+```
+_Note: at different viewport sizes, the message element hides the following elements:_
+- `<span class="o-message__detail">`
+- `<a class="o-message__action--secondary">`
 
-_Remember to start your codeblocks with three backticks and "html" so your markup is syntax highlighted correctly._
+A variation of the alert message is an **inner** alert message, which has almost exactly the same markup, with an optional addition of information, and does not have the option to close the message.
```

Comment:
A note here that no details are ever hidden on an inner message would be good, and maybe an explanation that an inner message is intended to sit within a page rather than span the top like a standard alert. Pedantic but might make it clearer :)